packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -543,16 +543,13 @@ contract RoutersFacet is BaseConnextFacet {
     if (_amount == 0) revert RoutersFacet__addLiquidityForRouter_amountIsZero();
 
     // Get the canonical asset ID from the representation.
-    (uint32 domain, bytes32 canonicalId) = s.tokenRegistry.getTokenId(_local);
-    bytes32 key = _calculateCanonicalHash(canonicalId, domain);
+    TokenId memory canonical = _getApprovedCanonicalId(_local);
```

Comment:
what is the difference between `_getApprovedCanonicalId` and `_getCanonicalTokenId`

Reply:
o i see, it also checks to see if it's been approved